packages/atomic/src/components/atomic-children-result-list/atomic-children-result-list.tsx

```diff
@@ -0,0 +1,64 @@
+import {
+  buildResultTemplatesManager,
+  FoldedResult,
+  ResultTemplatesManager,
+} from '@coveo/headless';
+import {Component, h, State, Element} from '@stencil/core';
+import {Bindings, InitializeBindings} from '../../utils/initialization-utils';
+import {TemplateContent} from '../atomic-result-template/atomic-result-template';
+
+import {FoldedResultContext} from '../result-template-components/result-template-decorators';
+
+@Component({
+  tag: 'atomic-children-result-list',
+  styleUrl: 'atomic-children-result-list.pcss',
+  shadow: true,
+})
+export class AtomicChildrenResultList {
+  private childResultsTemplatesManager!: ResultTemplatesManager<TemplateContent>;
+
+  @InitializeBindings() public bindings!: Bindings;
+  @Element() private host!: HTMLElement;
+
+  @FoldedResultContext() private foldedResult!: FoldedResult;
+
+  @State() public ready = false;
+  @State() public error!: Error;
+
+  private constructor() {
+    this.host.removeAttribute('result-children');
+  }
+
+  public async initialize() {
+    this.childResultsTemplatesManager = buildResultTemplatesManager(
+      this.bindings.engine
+    );
+    await this.registerChildrenResultTemplates();
+    this.ready = true;
```

Comment:
Instead of rendering twice using a state attribute, perhaps we could make `initialization-utils` await the `initialize` function. `initialization-utils` executes `initialize` inside [componentWillLoad](https://stenciljs.com/docs/component-lifecycle#componentwillload), which supports promises.